src/chttpd/src/chttpd_db.erl

```diff
@@ -825,7 +825,8 @@ db_doc_req(#httpd{method='COPY', user_ctx=Ctx}=Req, Db, SourceDocId) ->
         missing_rev -> nil;
         Rev -> Rev
     end,
-    {TargetDocId, TargetRevs} = couch_httpd_db:parse_copy_destination_header(Req),
+    {TargetDocId0, TargetRevs} = couch_httpd_db:parse_copy_destination_header(Req),
+    TargetDocId = list_to_binary(mochiweb_util:unquote(TargetDocId0)),
```

Comment:
I think the problem here is that the doc id isn't always escaped (that change is fairly recent).

Reply:
Thanks, Bob. According to statement from Garren in issue #977

```
Which means the API works although it doesn't follow 100% the Destination header's spec which states the header should contain an URI, hence its value should be encoded if it includes characters outside the valid range.
```

If spec implies that destination header for couchdb should be encoded, we need to decode here. If there is no escaped character in docid for target doc, calling `mochiweb_util:unquote/1` will not change the content of docid.